api/users/serializers.py

```diff
@@ -211,6 +211,12 @@ def update(self, instance, validated_data):
             elif 'accepted_terms_of_service' == attr:
                 if value and not instance.accepted_terms_of_service:
                     instance.accepted_terms_of_service = timezone.now()
+            elif 'region_id' == attr:
+                region_id = validated_data.get('region_id')
+                user_settings = instance._settings_model('osfstorage').objects.get(owner=instance)
+                user_settings.default_region_id = region_id
+                user_settings.save()
+                instance.default_region = self.context['request'].data['default_region']
```

Comment:
This feels weird to me, but I did it like this for speed.  So the user is already annotated with the original default_region id, so even though we're updating the region on UserSettings here, the original default_region annotation is still there and would be used.  So we need to manually override it somehow.

There are multiple ways to make sure this annotation is correct when your serialize the response, but pulling the default_region off of the original request didn't add any overhead regarding fetching the region again.